include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_utils.h

```diff
@@ -56,10 +56,14 @@ __device_info(const _ExecutionPolicy& __policy)
 #endif
 
 template <typename _ExecutionPolicy>
-::std::size_t
-__max_work_group_size(const _ExecutionPolicy& __policy)
-{
-    return __policy.queue().get_device().template get_info<sycl::info::device::max_work_group_size>();
+std::size_t
+__max_work_group_size(const _ExecutionPolicy& __policy, std::size_t __wg_size_limit = 8192)
```

Comment:
I am not sure that it's goo idea to specify max size on caller side.
As you described on this PR, we know the correct sizes for any devices.
Also we know that real incorrect info here has place only for FPGA_EMU device.
If it's correct, why we couldn't incorporate all this logic into this function?

Reply:
Additional to the correctness issues you mentioned, the caller might know a better limit than the max. This is used throughout our code with `min` comparisons, which can be eliminated with this change.